pandas/io/common.py

```diff
@@ -712,7 +713,17 @@ def get_handle(
 
     # Convert BytesIO or file objects passed with an encoding
     is_wrapped = False
-    if is_text and (compression or _is_binary_mode(handle, ioargs.mode)):
+    if ioargs.mode == "rb" and isinstance(handle, TextIOBase):
```

Comment:
Shouldn't this be
`if not is_text and ioargs.mode == "rb" and isinstance(handle, TextIOBase):`